SCloudRandomizer/MusicSource.m

```diff
@@ -64,24 +63,49 @@ - (void)getTracks:(SearchParams *)searchParams completionHandler:(tracksFetchedC
         completionHandler(tracks);
     };
     
-    // Replace spaces with '%20' and then replace commas with '%2C'
-    NSString *cleanedKeywords = [[searchParams.keywords stringByReplacingOccurrencesOfString:@" " withString:@"%20"] stringByReplacingOccurrencesOfString:@"," withString:@"%2C"];
-    NSString *resourceURL = [NSString stringWithFormat:@"https://api.soundcloud.com/tracks?format=json&q=%@", cleanedKeywords];
-    NSLog(@"The resourceURL is %@", resourceURL);
-    
     [SCRequest performMethod:SCRequestMethodGET
-                  onResource:[NSURL URLWithString:resourceURL]
+                  onResource:[self generateResourceURL:searchParams]
              usingParameters:nil
                  withAccount:[SCSoundCloud account]
       sendingProgressHandler:nil
              responseHandler:responseHandler];
 }
 
+- (NSURL *)generateResourceURL:(SearchParams *)params {
+    NSString *fromBpm = @"";
+    if (params.lowBpm > -1) {
+        fromBpm = [NSString stringWithFormat:@"&bpm[from]=%ld", (long)params.lowBpm];
+    }
+    
+    NSString *toBpm = @"";
+    if (params.highBpm > -1) {
+        toBpm = [NSString stringWithFormat:@"&bpm[to]=%ld", (long)params.highBpm];
+    }
+    
+    NSString *durationFrom = @"";
```

Comment:
Can you rename durationFrom to be something like durationFromInMilliSeconds or something so we know what that 60000 is down below?